python/cudf/cudf/core/dataframe.py

```diff
@@ -3801,14 +3801,15 @@ def agg(self, aggs, axis=None):
                 for key, value in aggs.items():
                     col = df_normalized[key]
                     result[key]= getattr(col, value)()
-            
+            else: raise ValueError("values of dict must be a string or list")
```

Comment:
Overall it seems a bit unwieldy to perform a loop over all the column names at every branch. Would it be better to create a single loop before all the branches and record the results with booleans?

Thoughts? @kkraus14

Reply:
I think the challenge is that you're handling a bunch of different kind of instance types for `agg` which makes it a bit challenging to deduplicate some of the code. I don't think it's an issue the way its written personally, but that's just my 2c.